### DISCUSSION TITLE
mvn clean install on main branch running into errors: Failed to start caches

### DISCUSSION THREAD

**Discussion by christian-2** [keycloak/keycloak#20376] [Q&A]
I have just tried building Keycloak from source with `mvn clean install` for the first time (as suggested in [`building.md`](https://github.com/keycloak/keycloak/blob/main/docs/building.md), with JDK 11.0.18, Maven 3.6.3 and the latest code base past 21.1.1 in `main`).

Apparently multiple tests in the category "Keycloak Quarkus Server Deployment" fail with this kind of pattern:

```
java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start caches
Caused by: java.util.concurrent.TimeoutException
```

Could it be that I am expected to install an external Infinispan server up front? But then nothing of this kind is mentioned in `building.md` nor [`CONTRIBUTING.md`](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md). Or is `building.md` perhaps outdated and one should refere to [`quarkus/README.md`](https://github.com/keycloak/keycloak/blob/main/quarkus/README.md) instead? How should I proceed?

**Comment by vmuzikar** [comment]
@christian-2 Thanks for reaching out and sorry you are facing issues while building Keycloak. Not quite sure what's happening there, please consider opening a GH Issue with a reproducer. In any case, you can skip the tests while building Keycloak by using `mvn clean install -DskipTests`.

**Reply by christian-2** [reply]
@vmuzikar FYI done #20494. Disabling the tests isn't really a good choice for me, because I was hoping to work on a PR (see #19216) that would require those tests before commit.
